data_steward/tools/create_tier.py

```diff
@@ -81,6 +83,93 @@ def get_dataset_name(tier, release_tag, deid_stage):
     return dataset_name
 
 
+def create_datasets(client, name, input_dataset, tier, release_tag):
+    """
+    Creates backup, staging, sandbox, and final datasets with the proper descriptions
+    and tag/labels applied
+
+    :param client: an instantiated bigquery client object
+    :param name: the base name of the datasets to be created
+    :param input_dataset: name of the input dataset
+    :param tier: tier parameter passed through from either a list or command line argument
+    :param release_tag: release tag parameter passed through either the command line arguments
+    :return: tuple of created dataset names
+    """
+
+    if not client:
+        raise RuntimeError("Please specify BigQuery client object")
+    if not name:
+        raise RuntimeError(
+            "Please specify the base name of the datasets to be created")
+    if not input_dataset:
+        raise RuntimeError("Please specify the name of the input dataset")
+    if not tier:
+        raise RuntimeError(
+            "Please specify the tier intended for the output datasets")
+    if not release_tag:
+        raise RuntimeError(
+            "Please specify the release tag for the dataset in the format of YYYY#q#r"
+        )
+
+    # Construct names of datasets need as part of the deid process
+    final_dataset_id = name
+    backup_dataset_id = f'{name}_{consts.BACKUP}'
+    staging_dataset_id = f'{name}_{consts.STAGING}'
+    sandbox_dataset_id = f'{name}_{consts.SANDBOX}'
+
+    dataset_ids = [
+        final_dataset_id, backup_dataset_id, staging_dataset_id,
+        sandbox_dataset_id
+    ]
+
+    # base labels and tags for the datasets
+    base_labels_and_tags = {'release_tag': release_tag, 'data_tier': tier}
+
+    description = f'Dataset created for {tier}{release_tag} CDR run'
+
+    # Creation of dataset objects
+    dataset_objects = []
+    for dataset in dataset_ids:
+        dataset_object = bq.define_dataset(client.project, dataset, description,
+                                           base_labels_and_tags)
+        dataset_objects.append(dataset_object)
+
+    # Creation of datasets
+    for dataset_object in dataset_objects:
+        client.create_dataset(dataset_object, exists_ok=False)
+
+    # Update the labels and tags
+    bq.update_labels_and_tags(sandbox_dataset_id, base_labels_and_tags,
+                              {'phase': {consts.SANDBOX}})
+    bq.update_labels_and_tags(backup_dataset_id, base_labels_and_tags, {
+        'de-identified': 'false',
+        'phase': {consts.BACKUP}
+    })
+    bq.update_labels_and_tags(staging_dataset_id, base_labels_and_tags, {
+        'de-identified': 'true',
+        'phase': {consts.STAGING}
+    })
+    bq.update_labels_and_tags(final_dataset_id, base_labels_and_tags,
+                              {'de-identified': 'true'})
```

Comment:
I'm not sure if the function `bq.update_labels_and_tags` works as intended, it runs into errors while calculating keys to update.

Reply:
You're right, it's not. When I ran my small integration test, only the original label and tags appeared. I looked in to it and `bq.update_labels_and_tags` only returns a dictionary of new or updated labels and tags, it doesn't actually apply them.

Reply:
I figured out an issue with `bq.update_labels_and_tags`. There was only one return statement and it was inside the `if overwrite_keys:` statement. This was making it so that it would never return the new dictionary of labels or tags if no keys were being overwritten. I added the same return statement outside of the `if overwrite_keys:` statement and it is now working. 

Now I need to figure out how to update the labels and tags in a loop since I will have to update them using `client.update_dataset`.

Reply:
If this is complete, let me know. `update_dataset` should be able to do this. Only existing keys need to be changed or new keys need to be added. The remaining keys should be left intact.

Reply:
Yes, this is working properly in the changes I pushed. I created a small integration test and made sure that it just added the new ones and the others remained.